experiments-core/src/foedus/ycsb/ycsb_driver.cpp

```diff
@@ -124,8 +127,27 @@ YcsbClientChannel* get_channel(Engine* engine) {
   return channel;
 }
 
-uint32_t YcsbClientChannel::peek_local_key_counter(uint32_t worker_id) {
-  return workers[worker_id].local_key_counter();
+// We put each worker's local key counter in shared memory, each occupying a cacheline
+// of bytes. So now the layout in the shared memory space is like:
+//
+// Offset: Content
+// 0: the channel
+// + align_up(sizeof(YcsbClientChannel), CACHELINE_SIZE): 1st worker's local key counter
+// + CACHELINE_SIZE: 2nd worker's local key counter
+// + CACHELINE_SIZE: 3rd worker's local key counter
+// ... and so on...
+uint32_t YcsbClientChannel::peek_local_key_counter(Engine* engine, uint32_t worker_id) {
+  return *get_local_key_counter(engine, worker_id);
+}
+
+uint32_t* get_local_key_counter(Engine* engine, uint32_t worker_id) {
```

Comment:
Just for readability, let's define a struct like:
```
  struct PerWorkerCounter {
    uint32_t key_counter_;
    /** padding to occupy its own cacheline. */
    char padding_[assorted::kCachelineSize - sizeof(uint32_t)];
  };
```
You can define it either in YcsbClientChannel as dummy-sized array with comments to warn about "actual size vs real size" or do something like below.